### PR TITLE
meow meow meow meow

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Comment by geoffjentry** [comment]
👍

File: src/main/scala/lenthall/ErrorOr.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+object ErrorOr {
+  type ErrorOr[A] = Validated[NonEmptyList[String], A]
+
+  implicit class FlatMap[A](val fa: ErrorOr[A]) extends AnyVal {
+    /**
+      * Not consistent with `Applicative#ap` but useful in for comprehensions.
+      *
+      * @see http://typelevel.org/cats/tut/validated.html#of-flatmaps-and-xors
+      */
+    def flatMap[B](f: A => ErrorOr[B]): ErrorOr[B] = {
```
Comment (geoffjentry) [inline src/main/scala/lenthall/ErrorOr.scala]: I thought the idea behind `core` (or to put another way, what I intended when I initially split the subprojects and made one called `core`) was to be stuff that exactly fits this description. i.e. a catch-all for things which should be shared across subprojects.

**Comment by kshakir** [comment]
ToL: test please ==> https://coveralls.io/builds/8079186/source?filename=src%2Fmain%2Fscala%2Flenthall%2FErrorOr.scala

File: src/main/scala/lenthall/ErrorOr.scala (hunk 1 above)
Comment (kshakir) [inline src/main/scala/lenthall/ErrorOr.scala]: ToL: (aka my 2¢) The GCS filesystem does not need to know about Workflows. cromwell.core seems like it contains a lot of common code for a workflow engine:

- `ExecutionStatus`
- `OutputStore`
- `DockerCredentials`
- `WdlValueSimpleton`
- etc.

Perfect for "a catch-all for things which should be shared across subprojects"… again of a workflow engine.

Meanwhile, lenthall is currently common utilities that _may be_ applied outside of a workflow engine, with artifact dependencies marked as `provided`:
- `AggregatedException`
- `SwaggerUiHttpService`
- `IntegerUtil`
- etc.

`ErrorOr` and it's custom flatMap to me fits this description. Side 2¢ (for free) I would give the FlatMap implicit a better name to define if it was error-accumulation and/or short-circuiting.

A sign that the dependencies are busted would be if the GCS file system artifact has a non-"provided" dependency on wdl4s when running [`sbt gcsFileSystem/dependencyGraph`](https://github.com/jrudolph/sbt-dependency-graph)

Comment (geoffjentry) [inline src/main/scala/lenthall/ErrorOr.scala]: We chatted about this in person a bit and punted to after standup tomorrow. What I said is that if there's something in core which shouldn't be accessible to the general system that to me this is an argument that that stuff shouldn't be in core, not to move stuff to another repo (a repo which IMO should be executed and then posthumously executed). I still view `core` as the place to stick stuff which might be useful throughout cromwell.

**Comment by mcovarr** [comment]
@kshakir per this morning's discussion, `ErrorOr` has been moved to Cromwell's `core` submodule.  I'll address the current lack of tests there.

**Comment by kshakir** [comment]
👍

**Pull request merged**